test/src/main/java/org/apache/accumulo/test/fate/zookeeper/ZooLockIT.java

```diff
@@ -306,6 +365,262 @@ public void testUnexpectedEvent() throws Exception {
+      List<String> children = zk1.getChildren(parent, false);
+      assertTrue(children.contains("zlock#00000000-0000-0000-0000-AAAAAAAAAAAA#0000000000"));
+      assertFalse("this node should have been deleted",
+          children.contains("zlock#00000000-0000-0000-0000-AAAAAAAAAAAA#0000000001"));
+      assertTrue(children.contains("zlock#00000000-0000-0000-0000-BBBBBBBBBBBB#0000000002"));
+      assertFalse("this node should have been deleted",
+          children.contains("zlock#00000000-0000-0000-0000-BBBBBBBBBBBB#0000000003"));
+
+      assertNull(zl1.getWatching());
+      assertEquals("/zlretry/zlock#00000000-0000-0000-0000-AAAAAAAAAAAA#0000000000",
+          zl2.getWatching());
+
+      zl1.unlock();
+      assertFalse(zlw1.isLockHeld());
+      zk1.close();
+
+      while (!zlw2.isLockHeld()) {
+        LockSupport.parkNanos(50);
+      }
+
+      assertTrue(zlw2.isLockHeld());
+      zl2.unlock();
+      zk2.close();
+
+    }
+
+  }
+
+  static class LockWorker implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LockWorker.class);
+
+    private final String parent;
+    private final String name;
+    private final CountDownLatch getLockLatch;
+    private final CountDownLatch lockCompletedLatch;
+    private final CountDownLatch unlockLatch = new CountDownLatch(1);
+    private final RetryLockWatcher lockWatcher = new RetryLockWatcher();
+    private Exception ex = null;
+
+    public LockWorker(final String parent, final String name, final CountDownLatch lockLatch,
+        final CountDownLatch lockCompletedLatch) {
+      this.parent = parent;
+      this.name = name;
+      this.getLockLatch = lockLatch;
+      this.lockCompletedLatch = lockCompletedLatch;
+    }
+
+    public void unlock() {
+      unlockLatch.countDown();
+    }
+
+    public boolean holdsLock() {
+      return lockWatcher.isLockHeld();
+    }
+
+    @Override
+    public void run() {
+      ConnectedWatcher watcher = new ConnectedWatcher();
+      try {
+        try (ZooKeeperWrapper zk =
+            new ZooKeeperWrapper(getCluster().getZooKeepers(), 30000, watcher)) {
+          zk.addAuthInfo("digest", "accumulo:secret".getBytes(UTF_8));
+          while (!watcher.isConnected()) {
+            Thread.sleep(50);
+          }
+          ZooReaderWriter zrw = new ZooReaderWriter(getCluster().getZooKeepers(), 30000, "secret") {
+            @Override
+            public ZooKeeper getZooKeeper() {
+              return zk;
+            }
+          };
+          ZooLock zl = new ZooLock(zrw, parent);
+          zl.setVMLockPrefix(new Prefix(name));
+          getLockLatch.countDown(); // signal we are done
+          getLockLatch.await(); // wait for others to finish
+          zl.lock(lockWatcher, "test1".getBytes(UTF_8)); // race to the lock
+          lockCompletedLatch.countDown();
+          unlockLatch.await();
+          zl.unlock();
+        }
+      } catch (IOException | InterruptedException | KeeperException e) {
```

Comment:
Will not catch runtime exceptions, would be good to catch those.

Reply:
will change to `Exception`